cmd/sealer/cmd/cluster/run-app.go

```diff
@@ -0,0 +1,144 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/sealerio/sealer/cmd/sealer/cmd/types"
+	"github.com/sealerio/sealer/common"
+	clusterruntime "github.com/sealerio/sealer/pkg/cluster-runtime"
+	"github.com/sealerio/sealer/pkg/clusterfile"
+	v12 "github.com/sealerio/sealer/pkg/define/image/v1"
+	"github.com/sealerio/sealer/pkg/define/options"
+	"github.com/sealerio/sealer/pkg/imagedistributor"
+	"github.com/sealerio/sealer/pkg/imageengine"
+	"github.com/sealerio/sealer/pkg/infradriver"
+	"github.com/sealerio/sealer/pkg/registry"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var appFlags *types.APPFlags
+var exampleForRunAppCmd = ``
+
+func NewRunAPPCmd() *cobra.Command {
+	runCmd := &cobra.Command{
+		Use:     "run-app",
+		Short:   "start to run an application cluster image",
+		Long:    `sealer run-app localhost/nginx:v1`,
+		Example: exampleForRunAppCmd,
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				cf              clusterfile.Interface
+				clusterFileData []byte
+				err             error
+			)
+
+			//todo grab more cluster info from api server
+			clusterFileData, err = ioutil.ReadFile(common.GetDefaultClusterfile())
+			if err != nil {
+				return err
+			}
+
+			cf, err = clusterfile.NewClusterFile(clusterFileData)
+			if err != nil {
+				return err
+			}
+
+			cluster := cf.GetCluster()
+			infraDriver, err := infradriver.NewInfraDriver(&cluster)
+			if err != nil {
+				return err
+			}
+
+			var (
+				clusterHosts = infraDriver.GetHostIPList()
+				launchCmds   = appFlags.LaunchCmds
+				appImageName = args[0]
+			)
+
+			imageEngine, err := imageengine.NewImageEngine(options.EngineGlobalConfigurations{})
+			if err != nil {
+				return err
+			}
+
+			extension, err := imageEngine.GetSealerImageExtension(&options.GetImageAnnoOptions{ImageNameOrID: appImageName})
+			if err != nil {
+				return fmt.Errorf("failed to get cluster image extension: %s", err)
+			}
+
+			if extension.Type != v12.AppInstaller {
+				return fmt.Errorf("exit install process, wrong cluster image type: %s", extension.Type)
+			}
+
+			clusterHostsPlatform, err := infraDriver.GetHostsPlatform(clusterHosts)
+			if err != nil {
+				return err
+			}
+
+			imageMounter, err := imagedistributor.NewImageMounter(imageEngine, clusterHostsPlatform)
+			if err != nil {
+				return err
+			}
+
+			imageMountInfo, err := imageMounter.Mount(appImageName)
+			if err != nil {
+				return err
+			}
+
+			defer func() {
+				err = imageMounter.Umount(imageMountInfo)
+				if err != nil {
+					logrus.Errorf("failed to umount cluster image")
+				}
+			}()
+
+			distributor, err := imagedistributor.NewScpDistributor(imageMountInfo, infraDriver, nil)
+			if err != nil {
+				return err
+			}
+
+			//todo grab this config from cluster file, that's because it belongs to cluster level information
+			var registryConfig registry.RegConfig
+			var config = registry.Registry{
```

Comment:
registry.Registry should be resolved from ClusterFile

Reply:
yes, registry config belongs to cluster level ，i already add todo for those logic.  I'm planning to update the cluster api after this PR merged.